src/cryptonote_core/tx_pool.cpp

```diff
@@ -254,39 +254,12 @@ namespace cryptonote
       return false;
     }
 
-    // fee per kilobyte, size rounded up.
     uint64_t fee;
 
-    if (tx.version == txversion::v1)
+    if (!get_tx_miner_fee(tx, fee, version >= HF_VERSION_FEE_BURNING))
```

Comment:
Do we allow constructing transactions that burn all the fee? Currently this allows txs where the burn fee >= tx fee to enter the mempool.

Reply:
`get_tx_miner_fee` always returns true for v2+ txes; it only returns false for v1 txes.  For modern txes it sets `fee` to the unspent amount in the tx (minus the burned amount, if present), then the `check_fee()` a few lines later checks it against the minimum required fee.

Reply:
m_fee_too_low shouldn't be set here if the failure case is it being an invalid V1 tx.

Reply:
To clarify: it returns false if the fee is too low *and* it's a v1 tx.